src/kw_remote.sh

```diff
@@ -43,24 +44,59 @@ function remote_main()
   fi
 }
 
-function list_remotes()
+# Returns the local or global remote.config location.
+# The local has precedence.
+function choose_correct_remote_config_file()
 {
-  local remote_config
-  local str_process
+  local has_local_config="false"
+  local has_global_config="false"
 
-  # Check if .kw folder exits
-  if [[ ! -d "${PWD}/.kw" ]]; then
-    complain 'You do not have .kw folder, try kw init first'
-    return 22 # EINVAL
+  if [[ -d "${PWD}/.kw" ]]; then
+    has_local_config="true"
+  fi
+
+  if [[ "${has_local_config}" = "false" && -d "${KW_ETC_DIR}" ]]; then
+    has_global_config="true"
   fi
 
-  # If we don't have a remote.config file yet, let the user know
-  if [[ ! -f "${local_remote_config_file}" ]]; then
-    complain 'Did you run kw init? It looks like that you do not have remote.config'
+  if [[ "${has_local_config}" = "true" ]]; then
+    #say "Using local remote.config." 1>&2
+    printf '%s' "${local_remote_config_file}"
+    return 0
+  elif [[ "${has_global_config}" = "true" ]]; then
+    #say "Using global remote.config." 1>&2
+    printf '%s' "${global_remote_config_file}"
+    return 0
+  else
+    complain 'No local or global config found. Please reinstall kw or run "kw init" to create a .kw in this dir.' 1>&2
     return 22 # EINVAL
   fi
+}
+
+function list_remotes()
+{
+  local remote_config
+  local str_process
+  local remote_config_file
+  local ret
+
+  remote_config_file="$(choose_correct_remote_config_file)"
+  ret="$?"
+  [[ "$ret" != "0" ]] && return "$ret"
+
+  if [[ ! -f "$remote_config_file" ]]; then
+    # If we don't have a remote.config file yet, let the user know
+    if [[ "$remote_config_file" = "$local_remote_config_file" ]]; then
```

Comment:
replace `=` with `-eq`.

Reply:
i'm getting some strange error when making this alteration.
`$> ./run_tests.sh test kw_remote >/dev/null`
`./src/kw_remote.sh: line 87: [[: /tmp/shunit.IQPPtY/tmp/.kw/remote.config: syntax error: operand expected (error token is "/tmp/shunit.IQPPtY/tmp/.kw/remote.config")`
`./src/kw_remote.sh: line 97: /tmp/shunit.IQPPtY/tmp/.kw/remote.config: No such file or directory`
`./src/kw_remote.sh: line 87: [[: /tmp/shunit.IQPPtY/tmp/.config/kw/remote.config: syntax error: operand expected (error token is "/tmp/shunit.IQPPtY/tmp/.config/kw/remote.config")`

Reply:
I reread the bash documentation, and I realized that -eq is only for arithmetic cooperation. We can use `=`, but tbh, I think `==` is more friendly since other languages use it; could you replace `=` with `==`?